src/App.tsx

```diff
@@ -15,33 +14,151 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setStarted] = useState(false);
+  const showGoods = () => {
+    setStarted(true);
+  };
+
+  const [sortMethod, setSortMethod] = useState(0);
+
+  const alphabetSort = () => {
+    setSortMethod(SortType.ALPABET);
+  };
+
+  const lengthSort = () => {
+    setSortMethod(SortType.LENGTH);
+  };
+
+  const [isReverse, setReverseMethod] = useState(false);
+  const reverseSort = () => {
+    setReverseMethod(current => !current);
+  };
+
+  const defaultSort = () => {
+    setSortMethod(SortType.NONE);
+    setReverseMethod(false);
+  };
+
+  const [minLength, setMinLength] = useState(1);
+  const minLengthSort = (n:number) => {
+    setMinLength(n);
+  };
+
+  function neededGoods(arr: string[]) {
+    const copy = [...arr];
+
+    copy.sort((a, b) => {
+      switch (sortMethod) {
+        case 1:
+          return a.localeCompare(b);
+
+        case 2:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
```

Comment:
Your sort will start every time and will check each element and do nothing

```suggestion
    switch (sortType) {
      case SortType.ALPABET:
        return visibleGoods.sort((a, b) => a.localeCompare(b));
      case SortType.LENGTH:
        return visibleGoods.sort((a, b) => a.length - b.length);
      case SortType.NONE:
      default:
        break;
    }
```

Reply:
fixed